tests/foreman/ui/test_ansible.py

```diff
@@ -602,6 +604,56 @@ def test_positive_ansible_config_report_changes_notice_and_failed_tasks_errors(
             assert f'notice Install the {package_name} package' in notice_log['permission_denied']
             assert f'Installed: rubygem-{package_name}' in notice_log['permission_denied']
 
+    @pytest.mark.no_containers
+    @pytest.mark.rhel_ver_list([settings.content_host.default_rhel_version])
+    def test_positive_debug_log_for_host(
+        self, target_sat, module_org, module_location, module_ak_with_cv
+    ):
+        """Verify debug mode output for host
+
+        :id: 06e0a95c-530b-11ef-a28c-36dcd2c0c415
+
+        :steps:
+            1. Enable Ansible Debug
+            2. Select two or more hosts
+            3. Run ansible roles on those hosts
+            4. Check the host execution logs
+
+        :expectedresults: The host execution log should show the debug log for expected host instead of the all host present in the inventory.
+
+        :BZ: 1912941
+
+        :Verifies: SAT-12267
+
+        :customerscenario: true
+        """
+        SELECTED_ROLE = 'RedHatInsights.insights-client'
+        nc = target_sat.nailgun_smart_proxy
+        nc.location = [module_location]
+        nc.organization = [module_org]
+        nc.update(['organization', 'location'])
+        target_sat.api.AnsibleRoles().sync(data={'proxy_id': nc.id, 'role_names': SELECTED_ROLE})
+        rhel_ver = settings.content_host.default_rhel_version
+        with Broker(nick=f'rhel{rhel_ver}', host_class=ContentHost, _count=2) as clients:
+            vm1, vm2 = clients
+            vm1.create_custom_repos(rhel8=settings.repos.rhel8_os)
+            result1 = vm1.register(module_org, module_location, module_ak_with_cv.name, target_sat)
+            assert result1.status == 0, f'Failed to register host: {result1.stderr}'
+            vm2.create_custom_repos(rhel8=settings.repos.rhel8_os)
+            result2 = vm2.register(module_org, module_location, module_ak_with_cv.name, target_sat)
+            assert result2.status == 0, f'Failed to register host: {result2.stderr}'
+            with target_sat.ui_session() as session:
+                session.organization.select(module_org.name)
+                session.location.select(module_location.name)
+                session.host.select_checkbox()
+                session.host._select_action('Run all Ansible roles', [vm1.hostname, vm2.hostname])
```

Comment:
Can you try using play_ansible_role method from ansible?

Reply:
I added `play_ansible_roles` here, but encountered a locator error because the `new_ui` tab wasn't populated. To resolve this, I used `@pytest.mark.parametrize('setting_update', ['lab_features=True'], indirect=True)` so that the `new_ui` tab displays correctly and avoids the locator error.